subprojects/docs/src/docs/userguide/publishingOverview.adoc

```diff
@@ -0,0 +1,262 @@
+    <sourcefile file="build.gradle" snippet="configure-generate-task"/>
+</sample>
+++++
+
+The above sample uses a regular expression to extract the name of the publication from the name of the task. This is so that there is no conflict between the file paths of all the POM files that might be generated. If you only have one publication, then you don't have to worry about such conflicts since there will only be one POM file.
+
+=== Terminology
+
+[[glossary:artifact]]
+Artifact::
+A file produced by a build, such as a JAR, a ZIP distribution, or a native executable.
++
+Artifacts are typically designed to be used or consumed by users or other projects, or deployed to hosting systems. This is distinct from intermediate files that are created by a build as a step on the way to producing the end result. For example, the class files generated by compiling Java source are almost never an end product themselves, since they are typically packaged into JARs or some other type of archive.
+
+[[glossary:component]]
+Component::
+A predefined combination of one or more artifacts and a standard set of information about those artifacts. The concept exists in Gradle only as a part of publishing.
++
+Components are typically defined by plugins and provide a simple way to define a publication, as they already include the appropriate metadata. For example, the `java` component consists of the production JAR — produced by the `jar` task — as well as its dependency information.
+
+[[glossary:configuration]]
+Configuration::
+A named collection of dependencies or artifacts.
++
+Gradle's configurations can be somewhat confusing because they apply to both <<sub:terminology_configuration,dependencies>> and artifacts. The main difference is that dependencies are consumed by the project, while artifacts are produced by it. Even then, the artifacts produced by a project are often consumed as dependencies by other projects.
++
+Configurations allow different aspects of the build to work with known subsets of a project's dependencies or artifacts, e.g. the dependencies required for compilation, or the artifacts related to a project's API.
+
+[[glossary:publication]]
+Publication::
+A description of the files and metadata that should be published to a repository as a single entity for use by consumers.
++
+A publication has a name, a primary artifact, information about that artifact — the exact information depends on the type of artifact and the repository it's published to — and optionally other, secondary artifacts, such as the source files used to produce the primary artifact or documentation on it.
+
+[[glossary:variant]]
+Variant::
+An alternative artifact for a <<sub:terminology_module,module>> that differs in some way from the main one, such as targetting a different platform or including debug information. Variants may have different dependencies from one another.
```

Comment:
Is a "component" distinct from a "module component"? Is a component something that can simply be consumed as a dependency or executed/run?

Reply:
A "module component" is a "component" provided by a module. As opposed to a "project component" which is a "component" provided by another subproject.

A component (or more precisely: one of its variants) is what you get when you resolve a dependency. What you execute is probably one of the artifacts.

Reply:
The project seen over its whole lifetime is a module indeed. We could certainly evolve those terms and end up with "ExternalComponent" and "ProjectComponent", both having a "ModuleIdentifier" giving you the id that holds over their lifetime. It's just not how it is right now. Right now we have "ModuleComponent" and "ProjectComponent" in our code base. It's confusing I know :)

Reply:
So should the definition of "module" include the fact that they are external to the projects that make up the build?

Also, "module version" seems to be the same thing as "component" based on [this definition](https://docs.gradle.org/current/userguide/dependency_management_terminology.html#sub:terminology_module_version). Should I make changes to the dependency management terminology chapter? BTW, I ultimately want to create a single glossary and remove that dependency management chapter.

Reply:
"module version" == "module component", yes. The term "module version" is an outdated part of the old resolution engine.

Reply:
To your first question: I don't know what we *should* do, that's partly up to the dependency management team. I can just tell you that right now the term "module" implies "external". But as you pointed out correctly, that doesn't really make that much sense and we should probably be talking about "External Component" and "Project Component", both of which being an instance of a "Module".

Reply:
I'm going to remove this section because the concept of variants isn't exposed to users right now. I think it will just cause confusion. As soon as the variant support is more explicit, we can document it properly.

 - [ ] Remove "variant" from the glossary